lib/webapp.js

```diff
@@ -134,6 +151,41 @@ DataserviceContext.prototype = {
   makeErrorObject: zluxUtil.makeErrorObject
 };
 
+function createDataserviceStorage(pluginId) {
+  
+  return process.clusterManager.getStorageAll(pluginId).then(function (clusterStorage) {
+
+    /* Get the whole plugin storage */
+    clusterStorage.getStorageAll = function () {
+      return process.clusterManager.getStorageAll(pluginId);
+    }
+
+    clusterStorage.getStorageByKey = function (key) {
+      return process.clusterManager.getStorageByKey(pluginId, key);
+    }
+
+    /* Set the whole plugin storage object */
+    clusterStorage.setStorageAll = function (dict) {
+      return process.clusterManager.setStorageAll(pluginId, dict);
+    }
+
+    clusterStorage.setStorageByKey = function (key, value) {
+      return process.clusterManager.setStorageByKey(pluginId, key, value);
+    }
+
+    clusterStorage.deleteStorageByKey = function (key) {
+      return process.clusterManager.deleteStorageByKey(pluginId, key);
+    }
+
+    contentLogger.debug("'" + pluginId + "' context is loaded with storage object: ", clusterStorage);
+    return clusterStorage;
+  })
+  .catch(function(e) { // Chances are, if storage failed to load, then the logger may have failed to load
```

Comment:
If you do somehow catch here, something bad has happened so the server should stop instead by sending a specific error number up to the cluster.